lib/plug/conn.ex

```diff
@@ -474,6 +477,11 @@ defmodule Plug.Conn do
                          "you have called send_chunked/2 before you send a chunk"
   end
 
+  defp iodata_empty?(""), do: true
+  defp iodata_empty?(binary) when is_binary(binary), do: false
+  defp iodata_empty?([]), do: true
+  defp iodata_empty?([hd | tl]), do: iodata_empty?(hd) && iodata_empty?(tl)
```

Comment:
For strictly boolean checks we should use `and`  instead of `&&`.
Also let's use `head` and `tail`.